wsmaster/che-core-sql-schema/src/main/resources/che-schema/6.11.0/1__add_signature_key_constraints.sql

```diff
@@ -0,0 +1,41 @@
+--
+-- Copyright (c) 2012-2018 Red Hat, Inc.
+-- This program and the accompanying materials are made
+-- available under the terms of the Eclipse Public License 2.0
+-- which is available at https://www.eclipse.org/legal/epl-2.0/
+--
+-- SPDX-License-Identifier: EPL-2.0
+--
+-- Contributors:
+--   Red Hat, Inc. - initial API and implementation
+--
+
+
+-- remove key pair records which linked to non-existing keys
+DELETE FROM che_sign_key_pair kp
+WHERE NOT EXISTS (
+   SELECT id
+   FROM   che_sign_key k
+   WHERE  k.id = kp.private_key
+   )
+OR NOT EXISTS (
+   SELECT id
+   FROM   che_sign_key k
+   WHERE  k.id = kp.public_key
+   );
+
+-- remove duplicated keys id if any
+DELETE FROM che_sign_key_pair kp1
+WHERE (
+      SELECT count(*) FROM che_sign_key_pair kp2
+      WHERE kp1.private_key  = kp2.private_key
+      OR kp1.private_key = kp2.public_key
+      ) > 1;
+
+
+-- add pair uniqueness constraint
+CREATE UNIQUE INDEX index_sign_public_private_key_id ON che_sign_key_pair (public_key, private_key);
+
+-- add keys table constraints
+ALTER TABLE che_sign_key_pair ADD CONSTRAINT fk_sign_public_key_id FOREIGN KEY (public_key) REFERENCES che_sign_key (id);
```

Comment:
Make sure that there are indexes for `fk_sign_public_key_id` and `fk_sign_private_key_id`

Reply:
Didn't get the idea. Explain, pls.

Reply:
>  We never do seatch on that fields

I would say that we don't do such searches explicitly. But I think JPA uses such search to fetch keys for a requested pair.

We have to add indexes on foreign keys, and we do it like here https://github.com/eclipse/che/blob/6c53555fded7621014501ab44d2c4bbca869b4db/wsmaster/che-core-sql-schema/src/main/resources/che-schema/5.8.0/1__add_foreigh_key_indexes.sql#L18-L17

Correct me If I'm wrong.

Reply:
Ok, i didn't remember the reason, but seems it's needed for  JPA. Will add separate indexes.